homeassistant/components/nuki/lock.py

```diff
@@ -147,3 +148,74 @@ def lock_n_go(self, unlatch=False, **kwargs):
         amount of time depending on the lock settings) and relock.
         """
         self._nuki_lock.lock_n_go(unlatch, kwargs)
+
+
+class NukiOpenerEntitiy(LockEntity):
+    """Representation of a Nuki opener."""
+
+    def __init__(self, nuki_opener):
+        """Initialize the opener."""
+        self._nuki_opener = nuki_opener
+        self._available = nuki_opener.state not in ERROR_STATES
+
+    @property
+    def name(self):
+        """Return the name of the opener."""
+        return self._nuki_opener.name
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        return self._nuki_opener.nuki_id
+
+    @property
+    def is_locked(self):
+        """Return true if ring-to-open (rto) is deactivated."""
+        return not self._nuki_opener.is_rto_activated
+
+    @property
+    def device_state_attributes(self):
+        """Return the device specific state attributes."""
+        data = {
+            ATTR_BATTERY_CRITICAL: self._nuki_opener.battery_critical,
```

Comment:
Do the openers have batteries?

Reply:
They do. They also have the option to plug them in.